apps/dashboard/src/components/primitives/table.tsx

```diff
@@ -3,23 +3,46 @@ import * as React from 'react';
 import { cn } from '@/utils/ui';
 import { ClassNameValue } from 'tailwind-merge';
 
-const Table = React.forwardRef<
-  HTMLTableElement,
-  React.HTMLAttributes<HTMLDivElement> & { containerClassname?: ClassNameValue }
->(({ className, containerClassname, ...props }, ref) => (
-  <div
-    className={cn(
-      'border-neutral-alpha-200 relative w-full overflow-x-auto rounded-md border shadow-sm',
-      containerClassname
-    )}
-  >
-    <table
-      ref={ref}
-      className={cn('relative w-full caption-bottom border-separate border-spacing-0 text-sm', className)}
-      {...props}
-    />
-  </div>
-));
+interface TableProps extends React.HTMLAttributes<HTMLDivElement> {
+  containerClassname?: ClassNameValue;
+  isLoading?: boolean;
+  loadingRowsCount?: number;
+  loadingRow?: React.ReactNode;
+}
+
+const LoadingRow = () => (
+  <TableRow>
+    <TableCell className="animate-pulse" colSpan={100}>
+      <div className="h-8 w-full rounded-md bg-neutral-100" />
+    </TableCell>
+  </TableRow>
+);
+
+const Table = React.forwardRef<HTMLTableElement, TableProps>(
```

Comment:
Added a default loading state to the table component

Reply:
Why it's needed if every table implementation is different?

Reply:
My thinking was to provide an easy way of doing loading states, unless more complex table structure exists